core/misc/autosubmit.js

```diff
@@ -39,9 +39,29 @@ Backdrop.behaviors.autosubmit = {
     // 'this' references the form element
     function triggerSubmit (e) {
       var $this = $(this);
+      // If this form is child of the view, not an independent block, make sure
+      // we can set focus back on the input field, after the whole view got
+      // replaced.
+      if ($this.parents('.view').length) {
+        $('body').attr('data-needs-focus', $this.find('.autosubmit-processed').attr('id'));
+      }
       $this.find('.autosubmit-click').trigger('click');
     }
 
+    // It's important to remove previous listeners, otherwise they stack up with
+    // every attach, triggered by typing.
+    $(document).off('ajaxStop').on('ajaxStop', function () {
```

Comment:
Question: would it be better to use `once()` instead of removing/attaching the listener?

Reply:
You are right. I did some testing - `once()` doesn't seem to be working on the `document` element, and `ajaxStop` only seems to work when attached to the document. So your code looks and works well.

I guess another alternative to tweaking the JS is to avoid replacing the exposed form upon ajax reloading.  But I haven't looked into this.